backend/tracim_backend/app_models/validator.py

```diff
@@ -1,11 +1,93 @@
+import re
+import typing
+
+from marshmallow import ValidationError
+from marshmallow.validate import Length
+from marshmallow.validate import Regexp
 from marshmallow.validate import OneOf
-from tracim_backend.app_models.contents import content_type_list
+from marshmallow.validate import Range
+from marshmallow.validate import Validator
 
+from tracim_backend.app_models.contents import GlobalStatus
+from tracim_backend.app_models.contents import content_status_list
+from tracim_backend.app_models.contents import content_type_list
 # TODO - G.M - 2018-08-08 - [GlobalVar] Refactor Global var
 # of tracim_backend, Be careful all_content_types_validator is a global_var !
+from tracim_backend.exceptions import TracimValidationFailed
+from tracim_backend.models.auth import Profile
+from tracim_backend.models.auth import User
+from tracim_backend.models.data import ActionDescription
+from tracim_backend.models.data import UserRoleInWorkspace
 
-all_content_types_validator = OneOf(choices=[])
+class TracimValidator(object):
+
+    def __init__(self):
+        self.validators = {}
+        self.values = {}
+
+    def add_validator(self, field:str, value:str, validator:typing.Callable) -> bool:  # nopep8
```

Comment:
You can be more precise on Callable type:

    validator:typing.Callable[[typing.Any], None]

Reply:
Can you add a doctsring ? One of my interrogation is for example: why when value is "false", the method return False ?

Reply:
@buxx 
That's because None case is not handled by validator for two main reason:
-> None case is not handled by validate in marshmallow (there is an "allow_none" param)
-> We want to test easily if all field are correct in user update, but update method allow none value, if value is none, param is not modified. It become really complex to deal with this if we handle None case in validator itself.